src/main/java/io/vertx/core/http/impl/Http1xClientConnection.java

```diff
@@ -479,15 +493,20 @@ private HttpClientResponseImpl beginResponse(HttpResponse resp) {
 
     private boolean endResponse(LastHttpContent trailer) {
       synchronized (conn) {
+        HttpClientRequestImpl req = request;
         if (conn.metrics != null) {
-          HttpClientRequestBase req = request;
           Object reqMetric = req.metric();
           if (req.exceptionOccurred != null) {
             conn.metrics.requestReset(reqMetric);
           } else {
             conn.metrics.responseEnd(reqMetric, response);
           }
         }
+        VertxTracer tracer = context.tracer();
+        if (tracer != null) {
+          List<Map.Entry<String, String>> tags = Collections.singletonList(new AbstractMap.SimpleEntry<>("http.status_code", "" + response.statusCode()));
```

Comment:
similarly here, not just data policy, but also we are allocating objects which may never be used. the tracer can decide what to do instead (certainly brave would)

Reply:
we're assuming here allocated objects are going to used by the tracer